### PR TITLE
Update terminator to run in a single account

### PR DESCRIPTION
**Pull request by abikouo**
This pull request contains the following updates:
- Align the branch with the latest updates from `mattclay/aws-terminator` since the repository was forked
- Update the terminator to run in a single account
- Remove the multiple environment support (switching stage between test and prod)
- Add a new feature to restrict the terminator to terminate only on a small set of resources

### CODE REVIEW

**Review by hakbailey — CHANGES_REQUESTED** [review]
We should probably update the `aws/README.md` file and the documentation in the Makefile to reflect the removal of stages. It would probably also be good to note that the terminator deployment playbook defaults to `terminate_small_set=true`, and what that means and how to change it.

File: aws/terminator_clean.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      amazon.aws.lambda_layer:
+        state: absent
+        name: "{{ lambda_layer_name }}"
+        version: -1
+
+    - name: Delete iam role for terminator functions
+      tags: iam
+      amazon.aws.iam_role:
+        name: "{{ iam_role_name }}"
+        state: absent
```
[the remark this answers is not in the document]
Comment (abikouo) [reply aws/terminator_clean.yml:41]: The role will be created with `create_instance_profile` set to `false`

**Review by hakbailey — APPROVED** [review]

File: aws/terminator/__init__.py [hunk 2]
```diff
@@ -72,18 +64,18 @@ def process_instance(instance: 'Terminator', check: bool, force: bool = False) -
...
     for terminator_type in sorted(get_concrete_subclasses(Terminator), key=lambda value: value.__name__):
-        if targets and terminator_type.__name__ not in targets:
+        if targets and terminator_type.__name__.lower() not in targets:
             continue
 
         # noinspection PyBroadException
         try:
             # noinspection PyUnresolvedReferences
-            instances = terminator_type.create(credentials)
+            instances = terminator_type.create()
```
Comment (gravesm) [inline aws/terminator/__init__.py:78]: Why remove the session here?

Comment (abikouo) [reply aws/terminator/__init__.py:78]: The current process needs to assume a Role from another AWS account, this means the current session in the lambda context is the one with the required permissions. With this new update the lambda session has the required perms, we no longer need to assume a role

File: aws/cleanup.py [hunk 3]
```diff
@@ -38,34 +72,24 @@ def main():
...
+                        required=True,
+                        help='The AWS region from which resources will be terminated')
 
+    parser.add_argument('--profile',
+                        required=True,
+                        help='The AWS profile')
 
-def parse_args():
-    parser = argparse.ArgumentParser(description='Terminate or destroy stale resources in the AWS test account.')
+    parser.add_argument('--table-name',
```
[the remark this answers is not in the document]
Comment (abikouo) [reply aws/cleanup.py:90]: ok. I will default to the `terminator_cleanup_resources` for the table name and give the possibility to override this value

**Pull request merged**